FlightsRepository.cs

```diff
@@ -0,0 +1,96 @@
+﻿using AirportTicketBooking.Models;
+
+namespace AirportTicketBooking.Repository
+{
+    public static class FlightsRepository
+    {
+        private const string FlightsFilePath = "Flights.csv";
+        private static bool _loaded = false;
+        private static List<FlightDTO> _flights = new();
+
+        public static List<FlightDTO> Flights
+        {
+            get
+            {
+                if (!_loaded && File.Exists(FlightsFilePath))
+                {
+                    _loaded = true;
+                    _flights = LoadFromFile(FlightsFilePath, true);
+                }
+
+                return _flights;
+            }
+        }
+
+        public static List<FlightDTO> LoadFromFile(string flightsFilePath, bool hasId = false)
+        {
+            if (!FlightValidator.IsValidFile(flightsFilePath, hasId))
+            {
+                throw new Exception($"Found errors at these lines in " +
+                    $"{Path.GetFileName(flightsFilePath)}:\n{FlightValidator.ErrorsLines}\n" +
+                    $"Make sure that you follow the constraints.");
+            }
+
+            var _fileReader = new StreamReader(flightsFilePath);
```

Comment:
Every stream needs to be closed, you need free the file and allow other processors to use it. 
A general rule, any class that inherits from `IDisposable` needs to be closed and disposed. and no need to call `.close()`

The way to do it here:
`using (var reader = new StreamReader(File.OpenRead(filePath))) {}`